include/contractor/contractor.hpp

```diff
@@ -1,1158 +1,73 @@
-#ifndef CONTRACTOR_HPP
-#define CONTRACTOR_HPP
+#ifndef PROCESSING_CHAIN_HPP
+#define PROCESSING_CHAIN_HPP
 
-#include "util/binary_heap.hpp"
-#include "util/deallocating_vector.hpp"
-#include "util/dynamic_graph.hpp"
-#include "util/percent.hpp"
+#include "extractor/edge_based_edge.hpp"
+#include "extractor/node_based_edge.hpp"
+#include "contractor/contractor.hpp"
+#include "contractor/contractor_config.hpp"
 #include "contractor/query_edge.hpp"
-#include "util/xor_fast_hash.hpp"
-#include "util/xor_fast_hash_storage.hpp"
-#include "util/integer_range.hpp"
-#include "util/simple_logger.hpp"
-#include "util/timing_util.hpp"
-#include "util/typedefs.hpp"
-
-#include <boost/assert.hpp>
-
-#include <stxxl/vector>
+#include "extractor/edge_based_edge.hpp"
+#include "util/static_graph.hpp"
+#include "util/deallocating_vector.hpp"
+#include "util/node_based_graph.hpp"
 
-#include <tbb/enumerable_thread_specific.h>
-#include <tbb/parallel_for.h>
-#include <tbb/parallel_sort.h>
+#include <boost/filesystem.hpp>
 
-#include <algorithm>
-#include <limits>
-#include <memory>
 #include <vector>
 
+struct lua_State;
+
 namespace osrm
 {
+namespace extractor
+{
+struct SpeedProfileProperties;
+struct EdgeBasedNode;
+struct EdgeBasedEdge;
+}
 namespace contractor
 {
 
+/// Base class of osrm-contract
```

Comment:
Tracking in https://github.com/Project-OSRM/osrm-backend/issues/1689#issuecomment-173275359 globally.